docs/debugging/cluster-reboot/main.go

```diff
@@ -0,0 +1,511 @@
+	if err != nil {
+		return
+	}
+
+	// bb, err := os.ReadFile("infra.json")
+	// if err != nil {
+	// 	panic(err)
+	// }
+	//
+	// err = json.Unmarshal(bb, &info)
+	// if err != nil {
+	// 	panic(err)
+	// }
+	//
+	pools = make(map[int]*Pool, 0)
+	for i := range info.Pools {
+		pools[i+1] = &Pool{
+			Servers: make(map[string]*Server, 0),
+		}
+	}
+
+	totalServers = 0
+	for _, s := range info.Servers {
+		if s.State == "offline" {
+			continue
+		}
+
+		totalServers++
+		pool := pools[s.PoolNumber]
+		server, ok := pool.Servers[s.Endpoint]
+		if !ok {
+			pool.Servers[s.Endpoint] = &Server{
+				Sets:     make(map[int]*Set, 0),
+				Rebooted: false,
+				Endpoint: s.Endpoint,
+			}
+			server = pool.Servers[s.Endpoint]
+		}
+
+		for _, d := range s.Disks {
+			set, ok := server.Sets[d.SetIndex]
+			if !ok {
+				server.Sets[d.SetIndex] = &Set{
+					Disks:      make(map[string]*Disk, 0),
+					SCParity:   info.Backend.StandardSCParity,
+					RRSCParity: info.Backend.RRSCParity,
+					ID:         d.SetIndex,
+					Pool:       d.PoolIndex,
+					CanReboot:  false,
+				}
+				set = server.Sets[d.SetIndex]
+			}
+
+			set.Disks[d.Endpoint] = &Disk{
+				UUID:           d.UUID,
+				Index:          d.DiskIndex,
+				Pool:           d.PoolIndex,
+				Server:         d.Endpoint,
+				Set:            d.SetIndex,
+				Path:           d.DrivePath,
+				State:          d.State,
+				UsedPercentage: (math.Ceil((float64(d.UsedSpace)/float64(d.TotalSpace)*100)*100) / 100),
+			}
+		}
+	}
+
+	return
+}
+
+func makeHostfile() {
+	pools, totalServers, err := getInfra()
+	var rebootRounds [200][200]map[string]*Server
+	processed := 0
+	for i := 0; i < len(rebootRounds); i++ {
+		if processed >= totalServers {
+			fmt.Printf("Total (%d) Online (%d)\n", totalServers, processed)
+			break
+		}
+
+		for pid, v := range pools {
+			if rebootRounds[i][pid] == nil {
+				rebootRounds[i][pid] = make(map[string]*Server)
+			}
+
+		nextServer:
+			for sid, s := range v.Servers {
+				if s.Processed {
+					continue
+				}
+
+				_, ok := rebootRounds[i][pid][s.Endpoint]
+				if !ok {
+
+					for _, rv := range rebootRounds[i][pid] {
+						if haveMatchingSets(rv, s) {
+							continue nextServer
+						}
+					}
+
+					rebootRounds[i][pid][s.Endpoint] = pools[pid].Servers[sid]
+					pools[pid].Servers[sid].Processed = true
+					processed++
+				} else {
+					continue
+				}
+
+			}
+		}
+	}
+
+	_ = os.RemoveAll(folder)
+	err = os.MkdirAll(folder, 0o777)
+	if err != nil {
+		panic(err)
+	}
+
+	var roundFile *os.File
+
+	for ri, rv := range rebootRounds {
+		for _, rv2 := range rv {
+			if rv2 != nil && len(rv2) > 0 {
+				roundFile, err = os.OpenFile(filepath.Join(folder, "round-"+strconv.Itoa(ri)), os.O_RDWR|os.O_CREATE|os.O_APPEND, 0o777)
+				if err != nil {
+					panic(err)
+				}
+				for _, rv3 := range rv2 {
+					// _, err = roundFile.WriteString(rv3.Endpoint + "\n")
+					_, err = roundFile.WriteString(strings.Replace(rv3.Endpoint, ":443", "", -1) + "\n")
```

Comment:
Same for `:80` ---- or any port???

Seems like you don't expect port to be on the host name in the file, correct?

Reply:
needs a split instead